src/features/domain-settings/hooks/useDomainSettingsForm.ts

```diff
@@ -0,0 +1,223 @@
+import { useCallback, useState } from 'react';
+
+import {
+	steps,
+	FormStep,
+	FieldValues,
+	DetailsFormSubmit,
+	ConfirmActionType,
+} from '../DomainSettings.types';
+import { useDomainSettingsData } from '.';
+import {
+	useIsDomainMetadataLocked,
+	useWeb3,
+	useZnsSdk,
+} from '../../../lib/hooks';
+import { useTransaction } from '@zero-tech/zapp-utils/hooks/useTransaction';
+
+import { Step } from '@zero-tech/zui/components';
+
+export type UseDomainSettingsFormFormReturn = {
+	stepId: string;
+	formHeader: string;
+	errorText: string;
+	confirmActionType: ConfirmActionType;
+	loadingStatusText: string;
+	isTransactionLoading: boolean;
+	onStepUpdate: (step: Step) => void;
+	onTitleUpdate: (title: string) => void;
+	onFormDetailsSubmit: (values: FieldValues) => void;
+	onLockMetadataStatus: () => void;
+	onSetAndLockMetadata: () => void;
+	onSetMetadata: () => void;
+	onConfirmActionUpdate: (action: ConfirmActionType) => void;
+};
+
+export const useDomainSettingsForm = (
+	zna: string,
+): UseDomainSettingsFormFormReturn => {
+	const sdk = useZnsSdk();
+
+	const { account, provider } = useWeb3();
+	const { executeTransaction } = useTransaction();
+
+	// add isLoading to form
+	const { domainId, metadata, metadataLockedStatus } =
+		useDomainSettingsData(zna);
+
+	const [stepId, setStepId] = useState(steps[0].id);
+	const [errorText, setErrorText] = useState<string>();
+	const [loadingStatusText, setLoadingStatusText] = useState<string>();
+	const [formHeader, setFormHeader] = useState<string>('My Domain Settings');
+	const [isTransactionLoading, setIsTransactionLoading] = useState<boolean>();
+	const [confirmActionType, setConfirmActionType] =
+		useState<ConfirmActionType>();
+
+	const [details, setDetails] = useState<DetailsFormSubmit>();
+	console.log(details);
+	console.log(metadata);
+
+	// Set metadata form details
+	const onFormDetailsSubmit = ({
+		title,
+		description,
+		isMintable,
+		isBiddable,
+		gridViewByDefault,
+		customDomainHeader,
+		customDomainHeaderValue,
+	}: FieldValues): void => {
+		setDetails({
+			attributes: metadata?.attributes,
+			title,
+			description,
+			image: metadata?.image,
+			image_full: metadata?.image_full,
+			previewImage: metadata?.previewImage,
+			animation_url: metadata?.animation_url,
+			stakingRequests: metadata?.stakingRequests,
+			isBiddable,
+			isMintable,
+			gridViewByDefault,
+			customDomainHeader,
+			customDomainHeaderValue,
+		});
+	};
+
+	const onStepUpdate = (step: Step) => {
+		setStepId(step.id);
+	};
```

Comment:
I think you may be able to skip these definitions, and just rename the `setState` functions in the hook return.

e.g.
```js
return {
    ...,
    onStepUpdate: setStepId,
    ...
}
```

Reply:
Removed these definitions.